bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/ModuleTypeResource.java

```diff
@@ -62,14 +64,15 @@
  */
 @Path("module-types")
 @Api("module-types")
+@NonNullByDefault
 @Component
 public class ModuleTypeResource implements RESTResource {
 
-    private ModuleTypeRegistry moduleTypeRegistry;
-    private LocaleService localeService;
+    private @NonNullByDefault({}) ModuleTypeRegistry moduleTypeRegistry;
```

Comment:
One general thing: I would prefer to use constructor injection instead of this annotation. But I would be fine to do it in a separate PR after this one has been merged.

Reply:
Yes this PR has become too big already so I'd like to wrap it up before I spent the rest of my time on resolving merge conflicts. :wink: 

I hope with having annotated all registries, any subsequent PR that adds null annotations will be easier.